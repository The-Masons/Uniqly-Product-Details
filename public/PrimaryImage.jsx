class PrimaryImage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      mouseOver: false,
    }
    this.moveImage = this.moveImage.bind(this);
  }

  moveImage(event) {
    if (this.state.mouseOver) {
      const tolerance = 80;
      const par = document.getElementById('primary-image-parent').getBoundingClientRect();
      const xDist = par.width;
      const yDist = par.height;
      let relX = -(event.clientX - par.left) + xDist / 2;
      let relY = -(event.clientY - par.top) + yDist / 2;

      document.getElementById('primary-image').style.transform = 'scale(2)';
      document.getElementById('primary-image').style.top = relY + 'px';
      document.getElementById('primary-image').style.left = relX + 'px';
    }
  }

  mouseExit(event){
    this.setState({
      mouseOver: false
    })
    document.getElementById('primary-image').style.transform = 'scale(1)';
    document.getElementById('primary-image').style.top = '0px';
    document.getElementById('primary-image').style.left = '0px';
  }

  render() {
    return (
      <div className="primary-image-parent" 
      id="primary-image-parent"
      onMouseMove={(event) => this.moveImage(event)}
      onMouseEnter={() => this.setState({mouseOver: true})}
      onMouseLeave={(event) => this.mouseExit(event)}>

        <img
        className="primary-image" 
        id="primary-image"
        src={this.props.image.img_url}/>
      </div>
    );
  }
}

export default PrimaryImage;