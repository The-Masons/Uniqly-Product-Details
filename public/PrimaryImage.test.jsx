import React from 'react';
import PrimaryImage from './PrimaryImage.jsx';

const image = "http://www.politifact.com/subjects/fake-news/"

describe('PrimaryImage', () => {
  test('should render with the correct items', () => {
    const primaryImage = mount(
      <PrimaryImage
        image={image}
        handleChangeImage={jest.fn()}
        />
      );
    expect(primaryImage).toMatchSnapshot();
  });

  test('should change mouseOver to true when image is hovered', () => {
    const primaryImage = mount(<PrimaryImage 
      image={image}
      handleChangeImage={jest.fn()}
    />);

    primaryImage.find('.primary-image-parent').simulate('mouseEnter')
  
    expect(primaryImage.state('mouseOver')).toEqual(true);
  });

  test('should change mouseOver to true when image is hovered', () => {
    const primaryImage = mount(<PrimaryImage 
      image={image}
      handleChangeImage={jest.fn()}
    />);

    primaryImage.find('.primary-image-parent').simulate('mouseLeave')
  
    expect(primaryImage.state('mouseOver')).toEqual(false);
  });

});