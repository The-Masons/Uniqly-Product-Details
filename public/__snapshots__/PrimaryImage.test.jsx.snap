// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryImage should render with the correct items 1`] = `
<PrimaryImage
  handleChangeImage={[MockFunction]}
  image="http://www.politifact.com/subjects/fake-news/"
>
  <div
    className="primary-image-section"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <img
      className="last-image-button"
      id="last-image-button"
      onClick={[Function]}
      src="back.png"
    />
    <div
      className="primary-image-parent"
      id="primary-image-parent"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
    >
      <img
        className="primary-image"
        id="primary-image"
      />
    </div>
    <img
      className="next-image-button"
      id="next-image-button"
      onClick={[Function]}
      src="back.png"
    />
  </div>
</PrimaryImage>
`;
